prisma/migrations/20251116172358_add_tests_module/migration.sql

```diff
@@ -0,0 +1,147 @@
+-- CreateEnum
+CREATE TYPE "QuestionType" AS ENUM ('MCQ', 'PROGRAMMING');
+
+-- CreateEnum
+CREATE TYPE "DifficultyLevel" AS ENUM ('EASY', 'MEDIUM', 'HARD');
+
+-- CreateTable
+CREATE TABLE "companies" (
+    "id" TEXT NOT NULL,
+    "name" TEXT NOT NULL,
+    "logo" TEXT,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "companies_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "tests" (
+    "id" TEXT NOT NULL,
+    "name" TEXT NOT NULL,
+    "description" TEXT NOT NULL,
+    "companyId" TEXT NOT NULL,
+    "difficulty" "DifficultyLevel" NOT NULL,
+    "duration" INTEGER NOT NULL,
+    "isActive" BOOLEAN NOT NULL DEFAULT true,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "tests_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "questions" (
+    "id" TEXT NOT NULL,
+    "testId" TEXT NOT NULL,
+    "type" "QuestionType" NOT NULL,
+    "text" TEXT NOT NULL,
+    "difficulty" "DifficultyLevel" NOT NULL,
+    "points" INTEGER NOT NULL,
+    "timeLimit" INTEGER,
+    "options" JSONB,
+    "programmingData" JSONB,
+    "order" INTEGER NOT NULL,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "questions_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "test_sessions" (
+    "id" TEXT NOT NULL,
+    "testId" TEXT NOT NULL,
+    "userId" TEXT NOT NULL,
```

Comment:
test_sessions defines a userId column but there is no foreign key to a users table. If users exist in the schema, this allows storing arbitrary userId values and breaks referential integrity.

**Sugerencia:**
```typescript
Add a foreign key constraint linking test_sessions.userId to users(id). Example:
ALTER TABLE "test_sessions" ADD CONSTRAINT "test_sessions_userId_fkey" FOREIGN KEY ("userId") REFERENCES "users"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
Priority: important (data integrity / safety)
```